### PR TITLE
Modlog: Avoid optional properties in ModlogEntry 

### PR DESCRIPTION
**Pull request by AnnikaCodes**

### CODE REVIEW

File: server/modlog.ts [hunk 1]
```diff
@@ -74,19 +74,21 @@ export interface ModlogSearch {
...
-	ip?: string;
-	isGlobal?: boolean;
-	loggedBy?: ID;
-	note?: string;
+	roomID: string;
+	visualRoomID: string;
+	userid: ID | null;
+	autoconfirmedID: ID | null;
+	alts: ID[];
+	ip: string;
```
Comment (monsanto) [inline server/modlog.ts:82]: Is `ip` always present? I see you `|| ''` elsewhere. If not, I would choose a strategy consistent with `userid`--empty string isn't a valid IP address.

**Review by monsanto — CHANGES_REQUESTED** [review]
Need corresponding `NOT NULL` entries in schema, otherwise looks good

File: server/modlog.ts (hunk 1 above)
Comment (AnnikaCodes) [reply server/modlog.ts:82]: no, it's not, i'll make it null

**Review by monsanto — APPROVED** [review]

**Comment by monsanto** [comment]
Until we have some migration support the procedure to hotpatch things with schema changes is set `Config.usesqlitemodlog = false`, delete `databases/modlog.db*`, hotpatch, `Config.usesqlitemodlog = true`

**Pull request merged**